### ISSUE TITLE
Open3 chdir parameter removes quoting

### ISSUE THREAD

**Issue by byteit101** [jruby/jruby#6565]
Passing quotes to Open3 are removed (and presumably parsed as multiple arguments) when using the `:chdir` parameter. This is different from MRI

**Environment Information**

JRuby 9.2 and master
Linux (Debian, x64)

**Expected Behavior (MRI)**

```
~ ruby -e "require 'open3';puts Open3.capture2('echo', '"'"hey"'"')[0];"       
"hey"
~ ruby -e "require 'open3';puts Open3.capture2('echo', '"'"hey"'"', chdir: '/')[0];"
"hey"
```

**Actual Behavior (JRuby)**

```
~ ruby -e "require 'open3';puts Open3.capture2('echo', '"'"hey"'"')[0];"       
"hey"
~ ruby -e "require 'open3';puts Open3.capture2('echo', '"'"hey"'"', chdir: '/')[0];"
hey

```
**Unquoted code**

```rb
require 'open3'
puts Open3.capture2('echo', '"hey"', chdir: '/')[0]
```

**Comment by byteit101** [comment]
Looks like this issue is related, particularly this comment: https://github.com/jruby/jruby/issues/985#issuecomment-319544506

**Comment by headius** [comment]
Are you sure this is broken on master? This could be a platform difference, but for me on MacOS it works:

```
$ ruby -e "require 'open3';puts Open3.capture2('echo', '"'"hey"'"', chdir: '/')[0];"
"hey"
```

I would have expected this to be fixed by @mrnoname1000's clever fix from #6226, which was specifically intended to preserve quoting when we have to use sh (which is the case here because of `chdir`).

**Comment by byteit101** [comment]
My bad, only on the latest released version and 2020-12-12 master. I was out of date

**Comment by headius** [comment]
Pushed #6568 to backport the fixes to 9.2.

**Comment by headius** [comment]
Fixed by #6568.
